doc/dev_guide/writing_extensions.rst

```diff
@@ -230,26 +230,31 @@ Equivalently, to add a new component 2D:
 Creating and registering new widgets and toolkeys
 -------------------------------------------------
 
-Toolkeys are functions to which it is possible to associate widgets. Extension
-can declare new toolkeys and widgets. For example, the `hyperspy-gui-traitsui
-<https://github.com/hyperspy/hyperspy_gui_traitsui>`_ and
-`hyperspy-gui-ipywidgets
-<https://github.com/hyperspy/hyperspy_gui_ipywidgets>`_ provide widgets for
-toolkeys declared in HyperSpy.
+To generate GUIs of specific method and functions, HyperSpy use widgets and toolkeys:
+
+* *widgets* are ipywidgets or traitsui objects, which can generate GUIs,
```

Comment:
That's not entirely true. In principle you could write a new widget "provider" in this way. Actually, that's what ``hyperspy-gui-traitsui`` and ``hyperspy-gui-ipywidgets`` do

Reply:
Indeed!